clang/tools/clang-scan-deps/ClangScanDeps.cpp

```diff
@@ -1024,11 +1024,17 @@ int clang_scan_deps_main(int argc, char **argv, const llvm::ToolContext &) {
             HadErrors = true;
         }
       } else if (ModuleName) {
+        StringRef NamesRef(*ModuleName);
+        SmallVector<StringRef, 16> NameList;
+        NamesRef.split(NameList, ",");
         auto MaybeModuleDepsGraph = WorkerTool.getModuleDependencies(
-            *ModuleName, Input->CommandLine, CWD, AlreadySeenModules,
-            LookupOutput);
-        if (handleModuleResult(*ModuleName, MaybeModuleDepsGraph, *FD,
-                               LocalIndex, DependencyOS, Errs))
+            NameList, Input->CommandLine, CWD,
+            AlreadySeenModules, LookupOutput);
+        // FIXME: Need to have better error handling logic for a list
+        // of modules. Probably through some call back.
```

Comment:
I don't think you need a callback. This should just drop the module name from the diag, as that should get printed by the normal Clang errors.